Neos.Flow/Classes/Validation/ValidatorResolver.php

```diff
@@ -296,9 +297,17 @@ protected function buildBaseValidatorConjunction($indexKey, $targetClassName, ar
             $conjunctionValidator->addValidator($objectValidator);
             foreach ($this->reflectionService->getClassPropertyNames($targetClassName) as $classPropertyName) {
                 $classPropertyTagsValues = $this->reflectionService->getPropertyTagsValues($targetClassName, $classPropertyName);
-
                 if (!isset($classPropertyTagsValues['var'])) {
-                    throw new \InvalidArgumentException(sprintf('There is no @var annotation for property "%s" in class "%s".', $classPropertyName, $targetClassName), 1363778104);
+                    try {
+                        $propertyReflection = new PropertyReflection($targetClassName, $classPropertyName);
+                    } catch (\ReflectionException $e) {
+                        throw new \RuntimeException(sprintf('Failed reflecting property %s from class %s while building base a validator conjunction: %s', $classPropertyName, $targetClassName, $e->getMessage()), 1651570561);
+                    }
+
+                    if (!$propertyReflection->hasType()) {
+                        throw new \InvalidArgumentException(sprintf('Failed building base validator conjunction for property %s in class %s because there is no @var annotation and no type declaration.', $classPropertyName, $targetClassName), 1363778104);
+                    }
+                    $classPropertyTagsValues['var'][] = $propertyReflection->getType()->getName();
```

Comment:
This should calm Psalm:

```suggestion
                    /** @var \ReflectionNamedType $type */
                    $type = $propertyReflection->getType();
                    $classPropertyTagsValues['var'][] = $type->getName();

```